File: Application/src/main/java/com/shareyourproxy/app/SearchActivity.java
package com.shareyourproxy.app;

import android.os.Bundle;
import android.support.v4.app.ActivityCompat;

import com.shareyourproxy.app.fragment.SearchFragment;

/**
 * Activity to handle displaying contacts and searching for new ones.
 */
public class SearchActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        preventStatusBarFlash(this);
        if (savedInstanceState == null) {
            SearchFragment searchFragment = SearchFragment.newInstance();
            getSupportFragmentManager().beginTransaction()
                .replace(android.R.id.content, searchFragment).commit();
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        ActivityCompat.finishAfterTransition(this);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/app/SearchActivity.kt
package com.shareyourproxy.app

import android.os.Bundle
import android.support.v4.app.ActivityCompat

import com.shareyourproxy.app.fragment.SearchFragment

/**
 * Activity to handle displaying contacts and searching for new ones.
 */
internal final class SearchActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        preventStatusBarFlash(this)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().replace(android.R.id.content, SearchFragment()).commit()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        ActivityCompat.finishAfterTransition(this)
    }
}
